hal/src/rtc/modes.rs

```diff
@@ -71,6 +71,14 @@ macro_rules! create_rtc_interrupt {
     };
 }
 
+// will be unused in boards that don't use syncbusy
```

Comment:
I only added the sync_busy() to Mode0, since that's all I'm using. I'm not even sure it's required, but it is Technically Correct.